Source/Core/DolphinQt2/Config/Graphics/PostProcessingConfigWindow.cpp

```diff
@@ -0,0 +1,359 @@
+// Copyright 2018 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "DolphinQt2/Config/Graphics/PostProcessingConfigWindow.h"
+
+#include <vector>
+
+#include <QCheckBox>
+#include <QDialog>
+#include <QDialogButtonBox>
+#include <QGridLayout>
+#include <QLabel>
+#include <QSlider>
+#include <QTabWidget>
+#include <QVBoxLayout>
+#include <QWidget>
+
+#include "VideoCommon/PostProcessing.h"
+#include "VideoCommon/RenderBase.h"
+#include "VideoCommon/VideoConfig.h"
+
+using ConfigurationOption = PostProcessingShaderConfiguration::ConfigurationOption;
+using OptionType = ConfigurationOption::OptionType;
+
+PostProcessingConfigWindow::PostProcessingConfigWindow(EnhancementsWidget* parent,
+                                                       const std::string& shader)
+    : QDialog((QWidget*)parent), m_shader(shader)
```

Comment:
This cast is likely unnecessary.

Reply:
I think iirc msvc was throwing an error, saying that the initialiser wouldn't accept it because it wasn't a concrete QWidget*

Reply:
Please check again. If it still doesn't work use ``static_cast<QWidget*>`` instead.

Reply:
If it complains of it not being a concrete type, it's because you've forward declared `EnhancementWidget` in the header, but didn't include the header file for it in the cpp file (so from the compiler's PoV, it can only see the forward declaration).